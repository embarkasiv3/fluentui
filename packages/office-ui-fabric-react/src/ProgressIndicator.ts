export * from './packages/react-fundamentals/ProgressIndicator';
