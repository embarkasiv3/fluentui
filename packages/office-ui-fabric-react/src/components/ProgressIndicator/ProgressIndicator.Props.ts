export interface IProgressIndicator {

}

export interface IProgressIndicatorProps {
  /**
   * Optional callback to access the IProgressIndicator interface. Use this instead of ref for accessing
   * the public methods and properties of the component.
   */
  componentRef?: (component: IProgressIndicator) => void;

  /**
   * Class name to apply to the root in addition to ms-ProgressIndicator.
   */
  className?: string;

  /**
   * Label to display above the control.
   */
  label?: string;

  /**
   * Text describing or supplementing the operation.
   */
  description?: string;

  /**
   * Percentage of the operation's completeness.
   */
  percentComplete?: number;

  /**
   * @deprecated
   * Deprecated at v0.43.0, to be removed at >= v0.53.0. Use 'label' instead.
   */
  title?: string;
}