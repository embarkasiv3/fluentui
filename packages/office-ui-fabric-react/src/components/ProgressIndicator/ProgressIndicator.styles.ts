import {
  FontSizes,
  FontWeights,
  HighContrastSelector,
  keyframes,
  noWrap,
} from '../../Styling';
import { getRTL } from '../../Utilities';
import {
  IProgressIndicatorStyleProps,
  IProgressIndicatorStyles,
} from './ProgressIndicator.types';

const IndeterminateProgress = keyframes({
  '0%': {
    left: '-30%',
  },
  '100%': {
    left: '100%',
  }
});
const IndeterminateProgressRTL = keyframes({
  '100%': {
    right: '-30%',
  },
  '0%': {
    right: '100%',
  }
});

export const getStyles = (
  props: IProgressIndicatorStyleProps
): IProgressIndicatorStyles => {
  const isRTL = getRTL();
  const {
    className,
    indeterminate,
    theme,
    barHeight = 2,
  } = props;

  const { palette, semanticColors } = theme;

  const marginBetweenText = 8;
  const textHeight = 18;

  return ({
    root: [
      'ms-ProgressIndicator',
      {
        fontWeight: FontWeights.regular,
      },
      className
    ],

    itemName: [
      'ms-ProgressIndicator-itemName',
      noWrap,
      {
        color: semanticColors.bodyText,
        fontSize: FontSizes.medium,
        paddingTop: marginBetweenText / 2,
        lineHeight: textHeight + 2,
      }
    ],

    itemDescription: [
      'ms-ProgressIndicator-itemDescription',
      {
        color: semanticColors.bodySubtext,
        fontSize: FontSizes.xSmall,
        lineHeight: textHeight,
      }
    ],

    itemProgress: [
      'ms-ProgressIndicator-itemProgress',
      {
        position: 'relative',
        overflow: 'hidden',
        height: barHeight,
        padding: `${marginBetweenText}px 0`,
      }
    ],

    progressTrack: [
      'ms-ProgressIndicator-progressTrack',
      {
        position: 'absolute',
        width: '100%',
        height: barHeight,
        backgroundColor: palette.neutralLight,

        selectors: {
          [HighContrastSelector]: {
            borderBottom: '1px solid WindowText',
          }
        }
      }
    ],

    progressBar: [
      'ms-ProgressIndicator-progressBar',
      {
        backgroundColor: palette.themePrimary,
        height: barHeight,
        position: 'absolute',
        transition: 'width .3s ease',
        width: 0,

        selectors: {
          [HighContrastSelector]: {
            backgroundColor: 'WindowText',
          }
        }
      },
      !indeterminate && {
        transition: 'width .15s linear'
      },
      indeterminate && [
        {
          position: 'absolute',
          minWidth: '33%',
          background: `linear-gradient(to right, transparent 0%, ${palette.themePrimary} 50%, transparent 100%)`,
          animation: `${IndeterminateProgress} 3s infinite`,
        },
        isRTL && { animation: `${IndeterminateProgressRTL} 3s infinite` },
      ],
    ],
  });
};