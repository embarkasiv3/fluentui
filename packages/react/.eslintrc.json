{
  // react-internal has some overrides we need to mirror
  "extends": ["../react-internal/.eslintrc"],
  "root": true
}
