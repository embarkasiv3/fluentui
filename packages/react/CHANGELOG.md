# Change Log - @fluentui/react

This log was last generated on Thu, 19 Mar 2020 17:52:34 GMT and should not be manually modified.
