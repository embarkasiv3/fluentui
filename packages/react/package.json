{
  "name": "@fluentui/react",
  "version": "8.0.0",
  "description": "Reusable React components for building web experiences.",
  "main": "lib-commonjs/index.js",
  "module": "lib/index.js",
  "typings": "lib/index.d.ts",
  "sideEffects": [
    "lib/version.js"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/microsoft/fluentui"
  },
  "license": "MIT",
  "scripts": {
    "build": "just-scripts build",
    "bundle": "just-scripts bundle",
    "clean": "just-scripts clean",
    "code-style": "just-scripts code-style",
    "codepen": "node ../../scripts/local-codepen.js",
    "just": "just-scripts",
    "lint": "just-scripts lint",
    "start": "just-scripts dev:storybook",
    "start:legacy": "yarn workspace @fluentui/public-docsite-resources start",
    "start-test": "just-scripts jest-watch",
    "test": "just-scripts test",
    "update-snapshots": "just-scripts jest -u",
    "mf": "just-scripts mf"
  },
  "devDependencies": {
    "@fluentui/eslint-plugin": "^1.0.0",
    "@types/chalk": "^2.0.0",
    "@types/enzyme": "3.10.3",
    "@types/enzyme-adapter-react-16": "1.0.3",
    "@types/glob": "^7.1.1",
    "@types/node": "^10.3.2",
    "@types/react": "16.9.42",
    "@types/react-dom": "16.9.10",
    "@types/react-test-renderer": "^16.0.0",
    "@types/webpack-env": "1.16.0",
    "@fluentui/scripts": "^1.0.0",
    "@fluentui/example-data": "^8.0.0",
    "@fluentui/jest-serializer-merge-styles": "^8.0.0",
    "@fluentui/test-utilities": "^8.0.0",
    "@fluentui/webpack-utilities": "^8.0.0",
    "@fluentui/react-conformance": "^0.2.1",
    "chalk": "^2.1.0",
    "enzyme": "~3.10.0",
    "enzyme-adapter-react-16": "^1.15.0",
    "glob": "^7.1.2",
    "jest-snapshot": "~24.9.0",
    "office-ui-fabric-core": "^11.0.0",
    "react": "16.8.6",
    "react-app-polyfill": "~1.0.1",
    "react-dom": "16.8.6",
    "react-test-renderer": "^16.3.0"
  },
  "dependencies": {
    "@fluentui/date-time-utilities": "^8.0.0",
    "@fluentui/foundation-legacy": "^8.0.0",
    "@fluentui/font-icons-mdl2": "^8.0.0",
    "@fluentui/merge-styles": "^8.0.0",
    "@fluentui/react-focus": "^8.0.0",
    "@fluentui/react-hooks": "^8.0.0",
    "@fluentui/react-shared-contexts": "^1.0.0-beta.9",
    "@fluentui/react-window-provider": "^2.0.0",
    "@fluentui/set-version": "^8.0.0",
    "@fluentui/style-utilities": "^8.0.0",
    "@fluentui/theme": "^2.0.0",
    "@fluentui/utilities": "^8.0.0",
    "@microsoft/load-themed-styles": "^1.10.26",
    "tslib": "^1.10.0"
  },
  "peerDependencies": {
    "@types/react": ">=16.8.0 <18.0.0",
    "@types/react-dom": ">=16.8.0 <18.0.0",
    "react": ">=16.8.0 <18.0.0",
    "react-dom": ">=16.8.0 <18.0.0"
  }
}
