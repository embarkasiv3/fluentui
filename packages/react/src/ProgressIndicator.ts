export * from '@fluentui/react-internal/lib/ProgressIndicator';
