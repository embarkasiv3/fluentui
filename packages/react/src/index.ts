import './version';
export * from '@fluentui/react-internal';
export * from './Breadcrumb';
export * from './Checkbox';
export * from './ComboBox';
export * from './DetailsList';
export * from './DocumentCard';
export * from './Dropdown';
export * from './GroupedList';
export * from './Link';
export * from './Pivot';
export * from './ShimmeredDetailsList';
export * from './Slider';
export * from './Toggle';
