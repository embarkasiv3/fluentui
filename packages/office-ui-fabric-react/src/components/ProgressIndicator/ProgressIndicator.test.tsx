import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { ProgressIndicator } from './ProgressIndicator';

describe('ProgressIndicator', () => {
  it('renders ProgressIndicator correctly', () => {
    const component = renderer.create(<ProgressIndicator />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});