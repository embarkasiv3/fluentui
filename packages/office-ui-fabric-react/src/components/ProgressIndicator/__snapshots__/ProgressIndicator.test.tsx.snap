// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressIndicator renders ProgressIndicator correctly 1`] = `
<div
  className="ms-ProgressIndicator"
>
  <div
    className="ms-ProgressIndicator-itemName"
  >
    
  </div>
  <div
    className="ms-ProgressIndicator-itemProgress"
  >
    <div
      className="ms-ProgressIndicator-progressTrack"
    />
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="0"
      aria-valuetext={undefined}
      className="ms-ProgressIndicator-progressBar"
      role="progressbar"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
  <div
    className="ms-ProgressIndicator-itemDescription"
  >
    
  </div>
</div>
`;
